spec/controllers/diy/file_yourself_controller_spec.rb

```diff
@@ -0,0 +1,98 @@
+require "rails_helper"
+
+RSpec.describe Diy::FileYourselfController do
+  describe "#edit" do
+    it "is 200 OK 👍" do
+      get :edit
+
+      expect(response).to be_ok
+    end
+  end
+
+  describe "#update" do
+    context "with valid params" do
+      let(:params) do
+        {
+          file_yourself_form: {
+            email_address: "example@example.com",
+            preferred_first_name: "Robot",
+            received_1099: "yes",
+            filing_frequency: "some_years",
+          }
+        }
+      end
+      before do
+        session[:source] = "beep"
+        session[:referrer] = "boop"
+        allow(subject).to receive(:visitor_id).and_return "blop"
+      end
+
+      context "without a diy intake in the session" do
+        it "creates a new diy intake and stores the id in the session" do
+          expect do
+            post :update, params: params
+          end.to change(DiyIntake, :count).by(1)
+
+          diy_intake = DiyIntake.last
+          expect(session[:diy_intake_id]).to eq diy_intake.id
+
+          expect(diy_intake.source).to eq "beep"
+          expect(diy_intake.referrer).to eq "boop"
+          expect(diy_intake.visitor_id).to eq "blop"
+          expect(diy_intake.locale).to eq "en"
+        end
+      end
+
+      context "with a diy intake in the session" do
+        let(:existing_diy_intake) { create :diy_intake, :filled_out, source: "existing_source", referrer: "existing_referrer", locale: "en", visitor_id: "existing_visitor_id" }
+        before do
+          session[:diy_intake_id] = existing_diy_intake.id
+        end
+
+        it "updates the intake from the session except source, referrer, etc" do
+          expect do
+            post :update, params: params
+          end.not_to change(DiyIntake, :count)
+
+          existing_diy_intake.reload
+          expect(existing_diy_intake.email_address).to eq "example@example.com"
+          expect(existing_diy_intake.preferred_first_name).to eq "Robot"
+          expect(existing_diy_intake.received_1099).to eq "yes"
+          expect(existing_diy_intake.filing_frequency).to eq "some_years"
```

Comment:
i got rid of the form mocking because it wasn't actually going to call any of the form methods with the double. this is a little redundant with the form spec but wasn't sure how else to check that it updated the right subset of params

Reply:
Yay